Fix: Add missing "types" field in package.json for proper TypeScript support

sntk000 opened the pull request:
### 📝 Pull Request description

<img width="845" alt="image" src="https://github.com/user-attachments/assets/7198c678-a672-475c-bd18-5e4025c5c11d" />


### Summary

This PR adds the missing `"types"` field in `package.json` to enable proper TypeScript support when importing the package.

Currently, when using `@purpom-media-lab/amplify-data-migration` in a TypeScript project, the following error may occur:

```

Cannot find module '@purpom-media-lab/amplify-data-migration' or its corresponding type declarations.ts(2307)

```

Even though the package provides `lib/index.d.ts`, it is not recognized by TypeScript unless explicitly referenced in `package.json`.

---

### Changes

- Added `"types": "lib/index.d.ts"` to the root `package.json`
- This enables editors (such as VSCode) and the TypeScript compiler to resolve the type declarations correctly
- and fix typo : [fix(typo): correct import path for amplify-data-migration in migration files](https://github.com/purpom-media-lab/amplify-data-migration/pull/117/commits/a20c57d609a93e5cb23cf6eabf7a705892097bb4)

---

### Impact

- Fixes the TypeScript import and IntelliSense issues
- Improves developer experience when creating migration files in TypeScript

---

### Note

If you plan to publish this package to npm in the future, this change ensures it works well in standard TypeScript projects without requiring additional configuration or workarounds.


### ✅ Verification

After adding the `"types"` field, the following has been verified:

* Installing `@purpom-media-lab/amplify-data-migration` via `npm` in a TypeScript project

* Running the following command to generate a migration file:

  ```bash
  npx data-migration create --name test_migration --migrationsDir migrations
  ```

* Importing types and modules from the package in the generated migration file (e.g. `Migration`, `MigrationContext`, etc.)

✅ **The TypeScript compiler no longer throws the following error:**

```
Cannot find module '@purpom-media-lab/amplify-data-migration' or its corresponding type declarations.ts(2307)
```

This confirms that the `"types"` field correctly resolves to the bundled type definitions at `lib/index.d.ts`.

sntk000 commented:
👋 Hi @fossamagna 

I'd like to request a review for this PR. It addresses a type resolution issue that occurs when importing the package in TypeScript projects.

### Summary

This PR adds a `types` field to the `package.json`, which enables editors and the TypeScript compiler to locate the correct declaration file (`lib/index.d.ts`). Without this field, the following error occurs in consumer projects:

```
Cannot find module '@purpom-media-lab/amplify-data-migration' or its corresponding type declarations.ts(2307)
```

### Verified

* Installed the package in a TypeScript project
* Ran `npx data-migration create --name test_migration`
* Successfully imported types (e.g. `Migration`, `MigrationContext`) in the generated migration file
* Confirmed that the TypeScript compiler no longer throws errors

Let me know if you have any feedback! 🙏

fossamagna reviewed (commented):
@sntk000 
Thanks for your contribution.
Could you please include changelog in this PR using `npx changeset`?

sntk000 commented:
> @sntk000 Thanks for your contribution. Could you please include changelog in this PR using `npx changeset`?

@fossamagna 
Thanks for the review!
I've added a changeset as requested:

> Type: patch
> Summary:
> Added missing "types" field to package.json to resolve TypeScript import errors.
> Also fixed a typo in import paths used in generated migration files.

c.f. [chore: update changeset]](https://github.com/purpom-media-lab/amplify-data-migration/pull/117/commits/85bdc7bb9b9b8c3d1bdd1de394c2ef582e04f6b1)

Let me know if any further adjustments are needed. 🙇‍♂️
```
npx changeset
🦋  What kind of change is this for @purpom-media-lab/amplify-data-migration? (current version is 1.2.0) · patch
🦋  Please enter a summary for this change (this will be in the changelogs).
🦋    (submit empty line to open external editor)
🦋  Summary · Added missing `"types"` field to `package.json` to resolve TypeScript import errors. Also fixed a typo in import paths used in generated migration files.
🦋  
🦋  === Summary of changesets ===
🦋  patch:  @purpom-media-lab/amplify-data-migration
🦋  
🦋  Is this your desired changeset? (Y/n) · true
🦋  Changeset added! - you can now commit it
🦋  
🦋  If you want to modify or expand on the changeset summary, you can find it here
🦋  info /Users/asaken_n47/Documents/Neurogica/amplify-data-migration/.changeset/dull-kangaroos-grow.md
```
